packages/koa-joi-validate/index.js

```diff
@@ -2,34 +2,42 @@ const { boomHelper } = require('@sigfox/koa-boom');
 const boom = require('boom');
 const Joi = require('joi');
 
-const koaJoiValidate = (schemasOrFunc, options = {}) => {
-  // @FIXME If schemasOrFunc is a Function, the unsupportedSchemas validation is useless.
-  const unsupportedSchemas = Object.keys(schemasOrFunc).filter(
+const throwOnUnsupportedSchemas = (schemas) => {
+  const unsupportedSchemas = Object.keys(schemas).filter(
     schema => !['body', 'headers', 'params', 'query'].includes(schema)
   );
   if (unsupportedSchemas.length > 0) {
     throw new Error(`Validation schemas for ${unsupportedSchemas.join(',')} are not supported!`);
   }
+};
+
+const koaJoiValidate = (schemasOrFunc, options = {}, preferredJoi) => {
```

Comment:
is the preferredJoi parameter only for testing purpose ?

Reply:
yes only way I found to be able to test multiple version of Joi.

Reply:
But it allows to change default joi version used by the package since we now support multiple version of Joi.